runtime/common/src/migrations.rs

```diff
@@ -896,6 +896,46 @@ where
     }
 }
 
+pub struct ExternalValidatorsInitialMigration<Runtime>(pub PhantomData<Runtime>);
+
+impl<Runtime> Migration for ExternalValidatorsInitialMigration<Runtime>
+where
+    Runtime: pallet_external_validators::Config,
+    Runtime: pallet_session::Config<
+        ValidatorId = <Runtime as pallet_external_validators::Config>::ValidatorId,
+    >,
+{
+    fn friendly_name(&self) -> &str {
+        "TM_ExternalValidatorsInitialMigration"
+    }
+
+    fn migrate(&self, _available_weight: Weight) -> Weight {
+        use frame_support::pallet_prelude::*;
+
+        // Set initial WhitelistedValidators to current validators from pallet session
+        let session_validators = pallet_session::Validators::<Runtime>::get();
```

Comment:
here we probably need to Session::queued_keys and get their map keys. The reason is because validatos are the current validators, queuedKeys contains the next ones, and i'ts the most up to date value

Reply:
fixed